### PR TITLE
Plausible Metadata

### PR DESCRIPTION
**Pull request by lewisgoddard**
### Changes Summary

- Move detectedOS to widget
- Add metadata to downloads
- Collate payments more sensibly

```
plausible('Downloads', {
  meta: {
    Region: config.user.region,
    Method: downloadMethod,
    OS: detectedOS(),
    Version: config.release.version
  }
})
```

Thanks to plausible/analytics#381

This pull request is ready for review.

### CODE REVIEW

File: _scripts/previous.js [hunk 1]
```diff
@@ -8,13 +8,26 @@
...
         // ACTION: .download-http.click: Track download over HTTP
         $('.download-link').click(function () {
-            plausible('Download of Previous Version')
-            plausible('Download of ' + config.previous.title + ' ' + config.previous.version)
+            if ($(this).hasClass('http')) {
+                var downloadMethod = 'HTTP'
+            }
+            if ($(this).hasClass('magnet')) {
+                var downloadMethod = 'Magnet'
+            }
```
Comment (btkostner) [inline _scripts/previous.js]:
```suggestion
			let downloadMethod = 'Unknown'

            if ($(this).hasClass('http')) {
                downloadMethod = 'HTTP'
            } else if ($(this).hasClass('magnet')) {
                downloadMethod = 'Magnet'
            }
```

File: _scripts/pages/download.js [hunk 2]
```diff
@@ -115,50 +116,44 @@ Promise.all([config, jQuery, Payment, modal]).then(([config, $, Payment]) => {
 
             // Free download
             if (Number.isNaN(paymentAmount) || paymentAmount < paymentMinimum) {
-                plausible('Payment: Skipped')
+                plausible('Payment', {
+                    meta: {
+                        Input: paymentAmount,
```
Comment (btkostner) [inline _scripts/pages/download.js:121]: I'm not sure we want Input here. We are going to end up with a list of mostly one events for every invalid value. I think `Amount` is good enough.

**Review by btkostner — CHANGES_REQUESTED** [review]
One thing and then it should be good

File: _scripts/pages/download.js (hunk 2 above)
Comment (lewisgoddard) [reply _scripts/pages/download.js:121]: I wanted to get the same info for all 3 payment results (Skip, Pay, Fail) to see what inputs were causing which action.

Comment (btkostner) [reply _scripts/pages/download.js:121]: Mhmm, alright. We will probably have a lot of noise though

**Review by btkostner — APPROVED** [review]

File: _scripts/previous.js [hunk 3]
```diff
@@ -8,13 +8,27 @@
...
-            plausible('Download of Previous Version')
-            plausible('Download of ' + config.previous.title + ' ' + config.previous.version)
+            let downloadMethod = 'Unknown'
+            if ($(this).hasClass('magnet')) {
+                downloadMethod = 'Magnet'
+            }
+            if ($(this).hasClass('http')) {
+                downloadMethod = 'HTTP'
+            }
+            plausible('Downloads', {
```
Comment (cassidyjames) [inline _scripts/previous.js:24]: `Download` makes more sense here and is what we have in Plausible dashboard.

**Pull request merged**